programs/stake/src/stake_instruction.rs

```diff
@@ -424,6 +425,41 @@ pub fn process_instruction(
                 Err(InstructionError::InvalidInstructionData)
             }
         }
+        Ok(StakeInstruction::Redelegate) => {
+            if invoke_context
+                .feature_set
+                .is_active(&feature_set::stake_redelegate_instruction::id())
+            {
+                let mut me = get_stake_account()?;
+                instruction_context.check_number_of_instruction_accounts(3)?;
+                let config_account =
+                    instruction_context.try_borrow_instruction_account(transaction_context, 3)?;
+                if !config::check_id(config_account.get_key()) {
+                    return Err(InstructionError::InvalidArgument);
+                }
+                let config =
+                    config::from(&config_account).ok_or(InstructionError::InvalidArgument)?;
+                drop(config_account);
+
+                redelegate(
+                    invoke_context,
+                    transaction_context,
+                    instruction_context,
+                    &mut me,
+                    1,
+                    2,
+                    &config,
+                    &signers,
+                )
+            } else {
+                if !invoke_context.feature_set.is_active(
+                    &feature_set::add_get_minimum_delegation_instruction_to_stake_program::id(),
+                ) {
+                    let _ = get_stake_account()?;
+                }
```

Comment:
I think we do need the logic. The only reason we don't need it is if we're willing to guarantee that `add_get_minimum_delegation_instruction_to_stake_program` is activated on all clusters before this software is rolled out, which seems 🤢 
Otherwise, a node that has upgraded to the software containing the Redelegate will hit the `Ok(StakeInstruction::Redelegate)` case, while one that hasn't will hit `Err(err)`, with potentially different behavior on an invalid stake account, if only one of those cases is calling `get_stake_account()?`

Reply:
Once `add_get_minimum_delegation_instruction_to_stake_program` is cleaned up, it should get less brittle...?

Reply:
@CriesofCarrots's comment "I think we do need the logic. ..." is exactly right, and is the reason why this yucky block is required. Assuming `add_get_minimum_delegation_instruction_to_stake_program` is enabled first is not a risk I wanted to take. It seemed better to include the extra code that could handle either activation order, and it won't be difficult for future cleanup to strip out this block once both features are activated.

Reply:
The reason why `StakeInstruction::DeactivateDelinquent` doesn't need this treatment is because that instruction landed *before* the `add_get_minimum_delegation_instruction_to_stake_program` feature.  So we're guaranteed that the behaviour of `StakeInstruction::DeactivateDelinquent` will be consistent across all combinations of `stake_deactivate_delinquent_instruction` and `add_get_minimum_delegation_instruction_to_stake_program` feature activations. 

Whereas `StakeInstruction::Redelegate` can't make that same claim because it hasn't landed yet.

Reply:
Actually, `StakeInstruction::DeactivateDelinquent` landed after `add_get_minimum_delegation_instruction_to_stake_program`, and I'm pretty confident you can use the pattern in that instruction handler here if you want to (which is what I was suggesting in [my very first comment in this thread](https://github.com/solana-labs/solana/pull/26294/files#r922747616)).
But I will reiterate the other part of that comment, that this is difficult to grok and super easy to screw up. I'll add some comments to help us along until `add_get_minimum_delegation_instruction_to_stake_program` is activated on all clusters.

Reply:
(All the other chaff from Brooks and me was us talking past each other but saying the same thing 😝 )

Reply:
ok, I pushed eacafe11615b243db1e3d167620f4a2d5d70472f to the PR for this (which I'll squash once I see CI is green).